examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,587 @@
+    X_test[mask_test.values],
+    df_train[mask_train],
+    df_test[mask_test],
+    target="AvgClaimAmount",
+    weights="ClaimNb",
+)
+print(scores)
+
+##############################################################################
+# Here, the scores for the test data call for caution as they are
+# significantly worse than for the training data indicating an overfit despite
+# the strong regularization.
+#
+# Note that the resulting model is the average claim amount per claim. As
+# such, it is conditional on having at least one claim, and cannot be used to
+# predict the average claim amount per policy in general.
+
+print("Mean AvgClaim Amount per policy:              %.2f "
+      % df_train["AvgClaimAmount"].mean())
+print("Mean AvgClaim Amount | NbClaim > 0:           %.2f"
+      % df_train["AvgClaimAmount"][df_train["AvgClaimAmount"] > 0].mean())
+print("Predicted Mean AvgClaim Amount | NbClaim > 0: %.2f"
+      % glm_sev.predict(X_train).mean())
+
+
+##############################################################################
+# We can visually compare observed and predicted values, aggregated for
+# the drivers age (``DrivAge``).
+
+fig, ax = plt.subplots(ncols=1, nrows=2, figsize=(16, 6))
+
+plot_obs_pred(
+    df=df_train.loc[mask_train],
+    feature="DrivAge",
+    weight="Exposure",
+    observed="AvgClaimAmount",
+    predicted=glm_sev.predict(X_train[mask_train.values]),
+    y_label="Average Claim Severity",
+    title="train data",
+    ax=ax[0],
+)
+
+plot_obs_pred(
+    df=df_test.loc[mask_test],
+    feature="DrivAge",
+    weight="Exposure",
+    observed="AvgClaimAmount",
+    predicted=glm_sev.predict(X_test[mask_test.values]),
+    y_label="Average Claim Severity",
+    title="test data",
+    ax=ax[1],
+    fill_legend=True
+)
+plt.tight_layout()
+
+##############################################################################
+# Overall, the drivers age (``DrivAge``) has a weak impact on the claim
+# severity, both in observed and predicted data.
+#
+# Pure Premium Modeling via a Product of Frequency and Severity
+# -------------------------------------------------------------
+# As mentioned in the introduction, the total claim amount per unit of
+# exposure can be modeled as the product of the prediction of the
+# frequency model by the prediction of the severity model.
+#
+# To quantify the aggregate performance of this product model, one can compute
+# the mean deviance of the train and test data assuming a Compound
+# Poisson-Gamma distribution of the total claim amount. This is equivalent to
+# a Tweedie distribution with "power" parameter between 1 and 2.
+#
+# As we do not know the true value of the "power" parameter, we compute the
+# mean deviances for a grid of possible values of the "power" parameter,
+# hoping that a good model for one value of "power" will stay a good model for
+# another. Here, every value of "power" defines a separate metric and models
+# are to be compared metric by metric:
```

Comment:
Thanks for your explanations on this @rth @lorentzenchr .

Do you think it would be better to compare the 2 estimators side by side as done in https://github.com/scikit-learn/scikit-learn/pull/14300#discussion_r386020130 ? I think it would make the comparison clearer and also avoid the duplication of the comments about using a grid of values for `power`. I'm happy to push something if that's OK

Reply:
I agree that putting results side by side would help (otherwise they are rather hard to read). Maybe concatenating the two frames along axis=1 and creating a multiindex (model name, test/train subset)?

Reply:
+1 on showing side by side.

Reply:
Cool, done in https://github.com/scikit-learn/scikit-learn/pull/14300/commits/293214c6dc78483e3802df8a9ac7b8ce5cede626